web/src/group_permission_settings.ts

```diff
@@ -123,3 +124,78 @@ export function get_realm_user_groups_for_dropdown_list_widget(
         };
     });
 }
+
+export type AssignedGroupPermission = {
+    setting_name: RealmGroupSettingName;
+    can_edit: boolean;
+    tooltip_message?: string;
+};
+
+export function get_assigned_permission_object(
+    setting_value: GroupSettingValue,
+    setting_name: RealmGroupSettingName,
+    group_id: number,
+    can_edit_settings: boolean,
+): AssignedGroupPermission | undefined {
+    const assigned_permission_object: AssignedGroupPermission = {
+        setting_name,
+        can_edit: can_edit_settings,
+    };
+
+    if (!can_edit_settings) {
+        assigned_permission_object.tooltip_message = $t({
+            defaultMessage: "You are not allowed to remove this permission.",
+        });
+    }
```

Comment:
Would it be more readable to return in this block? It seems like everything below has `can_edit_settings` guards, except the `undefined` cases, but I'm not sure how relevant those are to that code path. 

This function might benefit from a few more comments, and maybe some node tests.